x/profiles/types/msgs/msgs.go

```diff
@@ -104,3 +104,109 @@ func (msg MsgDeleteProfile) GetSignBytes() []byte {
 func (msg MsgDeleteProfile) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{msg.Creator}
 }
+
+// ----------------------
+// --- MsgRequestDTagTransfer
+// ----------------------
+
+// MsgRequestDTagTransfer define a Dtag transfer message
+type MsgRequestDTagTransfer struct {
+	CurrentOwner  sdk.AccAddress `json:"current_owner" yaml:"current_owner"`
+	ReceivingUser sdk.AccAddress `json:"receiving_user" yaml:"receiving_user"`
+}
+
+// NewMsgRequestDTagTransfer is a constructor function for MsgRequestDtagTransfer
+func NewMsgRequestDTagTransfer(owner, receiver sdk.AccAddress) MsgRequestDTagTransfer {
+	return MsgRequestDTagTransfer{
+		CurrentOwner:  owner,
+		ReceivingUser: receiver,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgRequestDTagTransfer) Route() string { return models.RouterKey }
+
+// Type should return the action
+func (msg MsgRequestDTagTransfer) Type() string { return models.ActionRequestDtag }
+
+func (msg MsgRequestDTagTransfer) ValidateBasic() error {
+	if msg.CurrentOwner.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid current owner address: %s", msg.CurrentOwner))
+	}
+
+	if msg.ReceivingUser.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid receiving user address: %s", msg.ReceivingUser))
+	}
+
+	if msg.ReceivingUser.Equals(msg.CurrentOwner) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "the receiving user and current owner must be different")
+	}
+
+	return nil
+}
+
+// GetSignBytes encodes the message for signing
+func (msg MsgRequestDTagTransfer) GetSignBytes() []byte {
+	return sdk.MustSortJSON(MsgsCodec.MustMarshalJSON(msg))
+}
+
+// GetSigners defines whose signature is required
+func (msg MsgRequestDTagTransfer) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.ReceivingUser}
+}
+
+// ----------------------
+// --- MsgAcceptDTagTransfer
+// ----------------------
+
+// MsgAcceptDTagTransfer represent a DTag transfer acceptance message
+type MsgAcceptDTagTransfer struct {
+	NewDTag       string         `json:"new_d_tag" yaml:"new_d_tag"`
```

Comment:
I think it's better to rename this `new_dtag` instead in order to be coherent with the namings all over

```suggestion
	NewDTag       string         `json:"new_d_tag" yaml:"new_dtag"`
```